webapp/channels/src/components/common/hooks/useAccessControlAttributes.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useState, useEffect, useCallback} from 'react';
+import {useDispatch} from 'react-redux';
+
+import type {AccessControlAttribute} from '@mattermost/types/access_control';
+
+import {getChannelAccessControlAttributes} from 'mattermost-redux/actions/channels';
+
+// Define supported entity types
+export enum EntityType {
+    Channel = 'channel',
+
+    // more entity types will be added here in the future
+}
+
+// Module-level cache for access control attributes
+// The cache stores data with a timestamp to implement a TTL (time-to-live)
+const attributesCache: Record<string, {
+    data: Record<string, string[]>;
+    timestamp: number;
+}> = {};
+
+// Cache TTL in milliseconds (5 minutes)
+const CACHE_TTL = 5 * 60 * 1000;
```

Comment:
IIRC, there's an in memory cache for ABAC stuff on server side as well. Does this TTL  match the cache TTL on server side for ABAC data? If not, we should match it. We should also add a link to the server side cache TTL constant here, and this constant's link on server side, so when someone updates one, they also remember to update the other.

Reply:
I do not think is necessary to match the TTL with the server cache since the intention of this piece of code is slightly different. This is not entirely for caching the values per se, is more for reducing the number of api calls, specially when we know we will most probably fetch a cached value. I do not have an stronger opinion of keeping it as it is now, but do not see neither a strong reason to match the server side cache TTL. Thoughts @isacikgoz ?